gcp_variant_transforms/testing/integration/run_tests.py

```diff
@@ -148,28 +146,68 @@ def _handle_failure(self, response):
             'No traceback. See logs for more information on error.')
 
   def validate_table(self):
-    """Runs a simple query against the output table and verifies aggregates."""
+    """Runs queries against the output table and verifies aggregates."""
     client = bigquery.Client(project=self._project)
-    # TODO(bashir2): Create macros for common queries and add the option for
-    # having a list of queries instead of just one.
-    query = self._validation_query.format(TABLE_NAME=self._table_name)
-    query_job = client.query(query)
+    # TODO(yifangchen): Create macros for common queries
+    query_formatter = QueryFormatter(self._table_name)
+    for assertion_config in self._assertion_configs:
+      _validate_assertion_config(assertion_config)
+      query = query_formatter.format_query(assertion_config['query'])
+      assertion = Assertion(client, query, assertion_config['expected_result'])
+      assertion.run_assertion()
+
+
+class Assertion(object):
+  """Runs a query and verifies that the output matches the expected result."""
+
+  def __init__(self, client, query, expected_result):
+    self._client = client
+    self._query = query
+    self._expected_result = expected_result
+
+  def run_assertion(self):
+    query_job = self._client.query(self._query)
     assert query_job.state == 'RUNNING'
     iterator = query_job.result(timeout=60)
     rows = list(iterator)
     if len(rows) != 1:
       raise TestCaseFailure('Expected one row in query result, got {}'.format(
           len(rows)))
     row = rows[0]
-    if len(self._expected_query_result) != len(row):
+    if len(self._expected_result) != len(row):
       raise TestCaseFailure(
           'Expected {} columns in the query result, got {}'.format(
-              len(self._expected_query_result), len(row)))
-    for key in self._expected_query_result.keys():
-      if self._expected_query_result[key] != row.get(key):
+              len(self._expected_result), len(row)))
+    for key in self._expected_result.keys():
+      if self._expected_result[key] != row.get(key):
         raise TestCaseFailure(
             'Column {} mismatch: expected {}, got {}'.format(
-                key, self._expected_query_result[key], row.get(key)))
+                key, self._expected_result[key], row.get(key)))
+
+
+class QueryFormatter(object):
+  """Formats a query.
+
+  Replaces keyword TABLE_NAME and eventually macros in the query.
+  """
+
+  def __init__(self, table_name):
+    self._table_name = table_name
+
+  def format_query(self, query):
+    """Formats the given ''query''.
+
+    Formatting logic is as follows:
+      Concatenates ''query'' parts into one string.
+      Replaces TABLE_NAME with the table associated for the query.
+
+    Args:
+      query (List[str]): a list of strings to be concatenated as one query.
```

Comment:
Instead of documenting the type in the 'Args:' section, please move them to a comment line after function definition, so line 197, 198 would like this:

`def format_query(self, query):`
`  # type: (List[str]) -> None`

This is the new style we are adopting but I have not managed to update the whole code yet (see Issue #108 for details).